openfe/setup/lomap_mapper.py

```diff
@@ -0,0 +1,52 @@
+# This code is part of OpenFE and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/openfe
+
+"""
+The MCS class from Lomap shamelessly wrapped and used here to match our API.
+
+"""
+import lomap
+
+
+from . import AtomMapper, AtomMapping
+
+
+class LomapAtomMapper(AtomMapper):
+    def __init__(self, time: int = 20, threed: bool = False,
+                 max3d: float = 1000.0):
+        """Wraps the MCS atom mapper from Lomap.
+
+        Kwargs are passed directly to the MCS class from Lomap for each mapping
+        created
+
+        Parameters
+        ----------
+        time : int, optional
+          timeout of MCS algorithm, passed to RDKit
+          default 20
+        threed : bool, optional
+          if true, positional info is used to choose between symmetry
+          equivalent mappings, default False
+        max3d : float, optional
+          maximum discrepancy in Angstroms between atoms before mapping is not
+          allowed, default 1000.0, which effectively trims no atoms
+        """
+        self.time = time
+        self.threed = threed
+        self.max3d = max3d
+
+    def _mappings_generator(self, mol1, mol2):
+        mcs = lomap.mcs.MCS(mol1, mol2, time=self.time, threed=self.threed,
+                            max3d=self.max3d)
+        # TODO: Once Lomap scorers exist, we'll want to keep a cache of
+        #       these mcs objects ({mapping: mcs}), so we can later query the
+        #       mcs that made a particular mapping to retrieve scores.
+
+        mapping_string = mcs.all_atom_match_list()
```

Comment:
This is quite ugly - can we upstream this and get a method to return a dictionary in lomap? Would be useful for anyone else using lomap in the future.

Reply:
possible yes

Reply:
issue raised